Source/WebKit/NetworkProcess/cache/NetworkCacheDataCurl.cpp

```diff
@@ -113,11 +113,14 @@ RefPtr<SharedMemory> Data::tryCreateSharedMemory() const
     if (isNull() || !isMap())
         return nullptr;
 
-    auto newHandle = std::get<FileSystem::MappedFileData>(*m_buffer).fileMapping();
-    if (!newHandle)
+    auto fileHandle = std::get<FileSystem::MappedFileData>(*m_buffer).fileMapping();
```

Comment:
> auto fileHandle = std::get<FileSystem::MappedFileData>(*m_buffer).fileMapping();

This code copies the handle.

> auto& fileHandle = std::get<FileSystem::MappedFileData>(*m_buffer).fileMapping();

This code doesn't copy the handle.


> auto fileHandle = std::get<FileSystem::MappedFileData>(*m_buffer).fileMapping().copy();

This code copies the handle only once.

Reply:
Yeah, exactly as fujii said.
We should remove `MachSendRight::copySendRight()` and `Win32Handle::copy()` to prevent this confusion.
So you could do something along the lines of:
```
  auto& fileData = std::get<FileSystem::MappedFileData>(*m_buffer);
  return SharedMemory::map({ fileData.fileMapping(), m_size }, SharedMemory::Protection::ReadOnly);
```